velox/functions/sparksql/String.h

```diff
@@ -1019,13 +1023,15 @@ struct ConvFunction {
     // When toBase is negative, converts to signed value. Otherwise, converts to
     // unsigned value. Overflow is allowed, consistent with Spark.
     if (toBase < 0) {
+      auto isMinInt64Num =
```

Comment:
I find this code confusing. Can you explain what's happening here? Maybe add a comment?

Can unsignedValue be less than min_int64? 

If min_int64 is a special value, then perhaps hard-code the result for that case and skip calling toChars altogether.

Reply:
> I find this code confusing. Can you explain what's happening here? Maybe add a comment?

This flag is used to make sure when we calculate the `resultSize` in `toChars` we always get a positive number. Due to the `std::abs(min_int64)` return the negative number, we need to make this special handle.

> Can unsignedValue be less than min_int64?

It can not less than min_int64, here we just check if the unsignedValue is equal to min_int64. The change history is here https://github.com/facebookincubator/velox/pull/8046#discussion_r1427510709

> If min_int64 is a special value, then perhaps hard-code the result for that case and skip calling toChars altogether.

Then we should maintain a hard-code list that map to the `toBase` [-36, -2], I'm not sure it is a good way as it requires more code and special value.